inductiva/_cli/cmd_logs/logs.py

```diff
@@ -1,22 +1,86 @@
 """CLI for logs."""
+import json
+import re
 import sys
+from typing import Tuple
 
 from .. import utils as cli_utils
-from ... import tasks
+from ... import tasks, ApiException
 
 
-def stream_task_logs(args):
-    """Consume the stream logs of a certain task."""
-    task_id = args.task_id
-    task = tasks.Task(task_id)
+def _get_task_id_from_mode(mode: str) -> Tuple[bool, str]:
+    """Extract the task_id from the mode."""
+    match = re.match(r"^(submitted|started)(-(\d+))?$", mode)
+    if not match:
+        return False, f"Invalid mode format: {mode}"
+
+    status_match = match.group(1)
+    offset_match = match.group(3)
+
+    status = status_match
+    offset = (int(offset_match) + 1) if offset_match else 1
+
+    # If the status is submitted, send None to the API to get all the tasks.
+    # The endpoint already returns the tasks in the submitted order by default
+    if status == "submitted":
+        status = None
+
+    task_list = tasks.get(last_n=offset, status=status)
+
+    if not task_list:
+        return False, f"No '{status}' tasks found."
+
+    return True, task_list[-1].id
+
+
+def validate_mode_or_task_id(mode: str) -> Tuple[bool, str]:
+    """Validate the mode or task_id."""
+    if mode.startswith("submitted") or mode.startswith("started"):
+        return _get_task_id_from_mode(mode)
+
+    if not cli_utils.is_task_id_valid(mode):
+        return False, f"Invalid task_id: {mode}"
+
+    return True, mode
 
-    if not task.is_running():
-        print(
-            f"The current status of task {task_id} is '{task.get_status()}'\n"
+
+def _check_if_task_is_running(task: tasks.Task) -> Tuple[bool, str]:
+    """Check if the task is running.
+    
+    This is done inside a try-except block to catch any exceptions that may
+    occur when trying to get the task status. Mainly if the task does not exist.
+    """
+
+    is_running = False
+    try:
+        is_running = task.is_running()
+    except ApiException as e:
+        return False, f"{ json.loads(e.body)['detail']}"
+
+    if not is_running:
+        return False, (
+            f"The current status of task {task.id} is '{task.get_status()}'\n"
```

Comment:
Occasionally, there might be a mismatch between `task.is_running()` and `task.get_status()` because these methods make 2 independent requests to the backend. I suggest you use the information returned by `task.get_info()`, but in that case it would be better to wait for #1456 to be merged.

Reply:
Now that #1456 is merged, I strongly suggest the use of `task.get_info()`. It avoids the possible mismatch between `task.is_running()` and `task.get_status()`, while it also reduces the number of API requests to 1.

```
    info = task.get_info()

    if not task.info.is_running:
        return False, (
            f"The current status of task {task.id} is '{task.info.status}'\n"
```

Reply:
Thanks Bruno! 
I've updated the code to use the get_info method!